src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -195,7 +195,9 @@ public object GetProperyValue(object obj, string propertyName)
                 var completed = await CompleteValue(context, context.Schema.FindType(nonNullType.Type), fields, result);
                 if (completed == null)
                 {
-                    throw new ExecutionError("Cannot return null for non-null type. Field: {0}".ToFormat(nonNullType.Name));
```

Comment:
Actually, looking at the applicable call sites, it seems like this will indeed work. See https://github.com/graphql-dotnet/graphql-dotnet/pull/32/files#diff-9ee075dfc3429298a3c5de0c1b53bef2R131 and all calls to `CompleteValue`.  Travelling atm, but will set up some unit tests for this later and verify.